### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by byapparov**

### CODE REVIEW

File: R/aws-s3.R [hunk 1]
```diff
@@ -1,32 +1,43 @@
...
+# Sys.setenv("AWS_ACCESS_KEY_ID" = "mykey",
+#            "AWS_SECRET_ACCESS_KEY" = "mysecretkey",
+#            "AWS_DEFAULT_REGION" = "us-east-1",
+#            "AWS_SESSION_TOKEN" = "mytoken")
+# See: https://github.com/cloudyr/aws.s3
+
+
+#' Saves the data frame to S3.
+#'
+#' @param df data to save
```
Comment (lmavridis) [inline R/aws-s3.R]: data.frame

File: R/aws-s3.R [hunk 2]
```diff
@@ -1,32 +1,43 @@
...
+#
+# Also aws.s3 package required following env variables:
+# Sys.setenv("AWS_ACCESS_KEY_ID" = "mykey",
+#            "AWS_SECRET_ACCESS_KEY" = "mysecretkey",
+#            "AWS_DEFAULT_REGION" = "us-east-1",
+#            "AWS_SESSION_TOKEN" = "mytoken")
+# See: https://github.com/cloudyr/aws.s3
+
+
+#' Saves the data frame to S3.
```
Comment (lmavridis) [inline R/aws-s3.R]: Saves a data.frame structure to S3

File: R/aws-s3.R [hunk 3]
```diff
@@ -1,32 +1,43 @@
...
-}
-
 
+#' Loads data from csv file in AWS S3
+#'
+#' @param path is the path to the S3 object
+#' @param header flag defines whether file has header
+#' @param bucket name of the bucket
+#' @param root project root path that is appended before the path
+#'
```
Comment (lmavridis) [inline R/aws-s3.R]: Since we agreed to work with data.tables should the s3GetFile return a data.table instead ?

File: R/aws-s3.R [hunk 4]
```diff
@@ -1,32 +1,43 @@
...
-  put_object(file = source.path, object = object.path, bucket = bucket)
-}
-
 
+#' Loads data from csv file in AWS S3
+#'
+#' @param path is the path to the S3 object
+#' @param header flag defines whether file has header
+#' @param bucket name of the bucket
+#' @param root project root path that is appended before the path
```
Comment (lmavridis) [inline R/aws-s3.R]: Keep one definition for the root and bucket throughout ?

File: R/bigquery.R [hunk 5]
```diff
@@ -85,15 +91,13 @@ createPartitionTable <- function(table, sql = NULL, file = NULL, existing.dates
   return (res)
 }
 
-
+#' Creates range teable in BigQuery.
+#'
+#' @param table name of the new table
+#' @param sql source for the table as string.
```
Comment (lmavridis) [inline R/bigquery.R]: Very pedantic comment but either (.) in all endings or in none ?

File: R/bigquery.R [hunk 6]
```diff
@@ -1,9 +1,14 @@
-library(bigrquery)
```
Comment (lmavridis) [inline R/bigquery.R]: some of the functions start with bq and some other not. Shall we be consistent ? And if yes shall we enforce it throughout the retl package ?

AWS s3 - s3Function
BigQuery - bqFunction
erp - erpFunction
etc. ?

File: R/aws-s3.R [hunk 7]
```diff
@@ -1,32 +1,43 @@
...
+#' @param bucket S3 bucket
+#' @param root Root folder that will be appended before the path
+s3PutFile <- function (df, file.name, path="",
                        bucket = Sys.getenv("AWS_S3_BUCKET"),
                        root = Sys.getenv("AWS_S3_ROOT")) {
-  # Saves the data frame in S3.
-  #   file.name: name of the file to be uploaded in S3.
-  #   path: S3 path starting after root folder, excludes filename. example: "folder/".
 
   write.csv(df, file = file.name, row.names = F)
```
Comment (daniel0000) [inline R/aws-s3.R]: This function does more than put a file into S3. Perhaps we should separate writing and uploading or add a parameter for filetype (csv, rds, ...).

File: R/aws-s3.R (hunk 2 above)
Comment (daniel0000) [inline R/aws-s3.R]: "Save input in S3 as a CSV."?

File: R/bigquery.R (hunk 6 above)
Comment (byapparov) [inline R/bigquery.R]: db / s3 / bq are aligned. Older functions in BigQuery file don't have a prefix, which we need to align with changes in dependant projects

File: R/aws-s3.R (hunk 7 above)
Comment (byapparov) [inline R/aws-s3.R]: it only writes csv to S3. there are specific functions for rds in aws.s3 package e.g. `aws.s3::s3saveRDS()`

**Comment by byapparov** [comment]
@lmavridis @daniel0000 Changes applied

File: R/aws-s3.R (hunk 7 above)
Comment (byapparov) [inline R/aws-s3.R]: I guess we could have a s3saveCSV() function, but because we only work with `csv` it will be redundant

Comment (lmavridis) [inline R/aws-s3.R]: Didn't you said we  should move towards rds ?

Comment (byapparov) [inline R/aws-s3.R]: Yes we should, totally. But this is just a clean up pull request, not new functionality or changes.

**Pull request merged**